games/twenty_forty_eight/heuristic.py

```diff
@@ -0,0 +1,53 @@
+import numpy as np
+
+from base.heuristic import Heuristic
+from games.twenty_forty_eight.state import TwentyFortyEightState
+from games.twenty_forty_eight.game import TwentyFortyEightGame
+
+MAX_TILE_CREDIT = 2048
+WEIGHT_MATRIX = np.array([
+    [2048, 1024, 64, 32],
+    [512, 128, 16, 2],
+    [256, 8, 2, 1],
+    [4, 2, 1, 1]
+])
+
+
+class TwentyFortyEightHeuristic(Heuristic[TwentyFortyEightState]):
+    def __init__(self, game: TwentyFortyEightGame) -> None:
+        self.game = game
+
+    def __call__(self, state: TwentyFortyEightState) -> float:
+        if self.game.is_terminal_state(state):
+            return np.inf * self.game.reward(state)
+        return self.empty_tiles(state) + self.max_tile_position(state) + self.weighted_board(state) \
+               - self.smoothness(state) + self._monotonicity(state)
+
+    def empty_tiles(self, state: TwentyFortyEightState):
+        return 512 * np.count_nonzero(state.board == 0)
+
+    def max_tile_position(self, state: TwentyFortyEightState):
+        max_tile = np.amax(state.board)
+        if state.board[0][0] == max_tile:
+            return MAX_TILE_CREDIT
+        else:
+            return -MAX_TILE_CREDIT
+
+    def weighted_board(self, state: TwentyFortyEightState):
+        return np.multiply(WEIGHT_MATRIX, state.board).sum()
```

Comment:
Z ciekawości, to czemu nie chciałaś użyć gwiazdki?
![obraz](https://user-images.githubusercontent.com/21079319/150153449-d3d76499-1ba3-4cd8-88e8-17fba18a3cf3.png)

Reply:
Hm w sumie to nie wiedziałam, że się da :sweat_smile: